src/data/set/intervals.lean

```diff
@@ -184,21 +188,34 @@ lemma Icc_subset_Iic_iff (h₁ : a₁ ≤ b₁) :
 lemma Icc_subset_Ici_iff (h₁ : a₁ ≤ b₁) :
   Icc a₁ b₁ ⊆ Ici a₂ ↔ a₂ ≤ a₁ :=
 ⟨λ h, h ⟨le_refl _, h₁⟩, λ h x ⟨hx, hx'⟩, le_trans h hx⟩
+
+@[simp] lemma Ici_inter_Iic : Ici a ∩ Iic b = Icc a b := rfl
+@[simp] lemma Ici_inter_Iio : Ici a ∩ Iio b = Ico a b := rfl
+@[simp] lemma Ioi_inter_Iic : Ioi a ∩ Iic b = Ioc a b := rfl
+@[simp] lemma Ioi_inter_Iio : Ioi a ∩ Iio b = Ioo a b := rfl
```

Comment:
I'm a little worried about the direction of these simp lemmas. You use them in the other direction below. Should they really be simp lemmas? The RHS is shorter, so in that sense, simpler.

Reply:
On the one hand, the RHS is shorter. On the other hand, the LHS uses simpler intervals. Maybe, I should remove `@[simp]`.

Reply:
Removed `@[simp]`.